### PR TITLE
Sigkill

### PR DESCRIPTION
**Pull request by DonDuala**
Cleaning up some event listening and refactored how bad oracle prices are handled

### CODE REVIEW

File: src/scalper.ts [hunk 1]
```diff
@@ -685,9 +686,10 @@ export class Scalper {
       (message: SerumVialTradeMessage) => {
         console.log(this.symbol, "Gamma Scalp Filled!", message.size, message.market, message.price, message.timestamp);
         let fillPrice = Number(message.price);
+        gammaScalpCount = gammaScalpCount + 1;
```
Comment (brittcyr) [inline src/scalper.ts:689]: generally better to not modify an input variable and either make a copy and modify that, or leave it as it was. a linter will probably complain about this.

File: src/scalper_utils.ts [hunk 2]
```diff
@@ -150,6 +150,33 @@ export function getDIPGamma(
...
+        fairValue,
+        dip.strike,
+        yearsUntilMaturity,
+        impliedVol,
+        rfRate,
+        dip.type
+      ) *
+        dip.qty +
+        thetaSum;
+        thetaSum = thetaSum;
```
Comment (brittcyr) [inline src/scalper_utils.ts:175]: what is this line for?

File: src/scalper.ts [hunk 3]
```diff
@@ -510,7 +510,8 @@ export class Scalper {
     catch (err){
       console.log(this.symbol, "Main Gamma Error", err, err.stack)
     }
-    this.serumVialClient.removeAnyListeners();
```
Comment (brittcyr) [inline src/scalper.ts]: why remove this?

File: src/scalper.ts [hunk 4]
```diff
@@ -795,3 +803,16 @@ function waitForFill(conditionFunction) {
   };
   return new Promise(poll);
 }
+
+function waitForGamma(conditionFunction) {
```
Comment (brittcyr) [inline src/scalper.ts:807]: can you put a comment above this to explain what it is doing? Seems like it is waiting for either all the gamma scalps to happen, a sufficient amount of polls?

**Review by brittcyr — APPROVED** [review]

File: src/scalper.ts (hunk 1 above)
Comment (DonDuala) [reply src/scalper.ts:689]: Wanted to confirm that it gets picked up in waitForGamma. Seemed easier to manage just the 1 counter variable for that

**Pull request merged**